src/nix/ls.cc

```diff
@@ -145,7 +145,11 @@ struct CmdLsNar : Command, MixLs
 
     void run() override
     {
-        list(makeNarAccessor(readFile(narPath)), CanonPath{path});
+        AutoCloseFD fd = open(narPath.c_str(), O_RDONLY);
+        auto source = FdSource{fd.get()};
+        auto narAccessor = makeNarAccessor(source);
+        auto listing = listNar(narAccessor, CanonPath::root, true);
+        list(makeLazyNarAccessor(listing, seekableGetNarBytes(narPath)), CanonPath{path});
```

Comment:
I really tried to remove the std::optional<string> from NarAccessor but it's used by RemoteFSAccessor
I think we can make RemoteFSAccessor a lot faster by also using Sink & Source and not storing the whole memory of the NAR